src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -6,79 +6,109 @@
  * @since 18.6.2017
  */
 public class Tree {
-    private int value;
```

Comment:
I will try. Made it this way because I was struggled with the previous problem and I did not solved it. Tried to make class Node to expands class Tree but there was a problem with the constructors, which I did not found a solution.